packages/wallets/src/services/wallet-factory.ts

```diff
@@ -173,12 +189,75 @@ export class WalletFactory {
         }
     }
 
+    /**
+     * Checks if the wallet config passed during wallet initialization matches the API response
+     * Throws on mismatch
+     * @param existingWallet Onchain wallet state
+     * @param args Wallet config passed during wallet initialization
+     */
+    private checkWalletConfig<WalletType extends keyof WalletTypeToArgs>(
+        existingWallet: GetWalletSuccessResponse,
+        args: WalletTypeToArgs[WalletType]
+    ) {
+        switch (existingWallet.type) {
+            case "evm-smart-wallet": {
+                const { adminSigner: adminSignerInput } = args as WalletTypeToArgs["evm-smart-wallet"];
```

Comment:
the code is the same for keypairs so we can unify for both evm and solana smart wallets

Reply:
https://github.com/Crossmint/crossmint-sdk/pull/1009/commits/18f1956a88869cb63e2d2d82d173db4a1be12230